DSharpPlus.Caching/Abstractions/ICacheService.cs

```diff
@@ -0,0 +1,39 @@
+using System.Threading.Tasks;
+
+namespace DSharpPlus.Caching.Abstractions
+{
+    /// <summary>
+    /// Exposes API to access caches without needing to directly rely on IMemoryCache, IDistributedCache or the likes.
+    /// </summary>
+    public interface ICacheService
```

Comment:
This interface is devoid of eviction methods, as well as a method of retrieving evicted data.

Reply:
the interface now contains a `ValueTask<TItem?> RemoveAsync<TItem?>(object)` method, which evicts the cache entry and returns the cached data under the given key